File: content/problems/2685-count-the-number-of-complete-components/solutions/component-vertex-and-degree-count/solution.cpp
class Solution {
    public:int countCompleteComponents(int n,vector<vector<int>>& edges){
        vector<vector<int>> graph(n);
        for(auto& edge:edges){
            graph[edge[0]].push_back(edge[1]);
            graph[edge[1]].push_back(edge[0]);
        }
        vector<bool> visited(n);
        int answer=0;
        for(int start=0;start<n;++start)if(!visited[start]){
            queue<int> pending;
            pending.push(start);
            visited[start]=true;
            int vertices=0,degreeSum=0;
            while(!pending.empty()){
                int node=pending.front();
                pending.pop();
                ++vertices;
                degreeSum+=graph[node].size();
                for(int next:graph[node])if(!visited[next]){
                    visited[next]=true;
                    pending.push(next);
                }
            }
            answer+=degreeSum==vertices*(vertices-1);
        }
        return answer;
    }
};

File: content/problems/2685-count-the-number-of-complete-components/solutions/component-vertex-and-degree-count/Solution.kt
class Solution {
    fun countCompleteComponents(n:Int,edges:Array<IntArray>):Int{
        val graph=Array(n){
            mutableListOf<Int>()
        };
        for(edge in edges){
            graph[edge[0]].add(edge[1]);
            graph[edge[1]].add(edge[0])
        };
        val visited=BooleanArray(n);
        var answer=0;
        for(start in 0 until n)if(!visited[start]){
            val queue=java.util.ArrayDeque<Int>();
            queue.add(start);
            visited[start]=true;
            var vertices=0;
            var degreeSum=0;
            while(queue.isNotEmpty()){
                val node=queue.removeFirst();
                vertices++;
                degreeSum+=graph[node].size;
                for(next in graph[node])if(!visited[next]){
                    visited[next]=true;
                    queue.add(next)
                }
            };
            if(degreeSum==vertices*(vertices-1))answer++
        };
        return answer
    }
}
